### PR TITLE
[BL602] Update README, Fix compile, Fix pbuf length

### PR DESCRIPTION
**Pull request by jczhang777**
#### Problem

* Update README, Added supported Demo boards, flash tools addr, and chip-tool control commands.
* Fix bl602 lwip pbuf alloc size is not matching matter pbuf size.
* Fix compile error.

#### Change overview
This pr is about update readme, fix compile error and fix bl602 pbuf alloc size.

#### Testing

* Fetch the code, and compile lighting app, flash it. Use chip-tool to control it.
Use chip-tool commission a devive. After successful commissioning, use the OnOff cluster commands to control
the OnOff attribute. These operations can be performed correctly

### CODE REVIEW

**Review by andy31415 — DISMISSED** [review]
Please update summary with full information:

- delete boilerplate (`What is being fixed? Examples:` seems not relevant as a text)
- `Fix pbuf issue` - what issue?
- `Update README` with what? for what purpose?

Also add "how was this tested" section.

File: examples/lighting-app/bouffalolab/bl602/README.md [hunk 1]
```diff
@@ -1,90 +1,95 @@
...
-         ```
+         connectedhomeip/out/bl602-light/chip-bl602-lighting-example.bin
 
 -   To delete generated executable, libraries and object files use:
 
-         ```
-         $ cd ~/connectedhomeip/
-         $ rm -rf out/
-         ```
+          $ cd ~/connectedhomeip/
```
Comment (andy31415) [inline examples/lighting-app/bouffalolab/bl602/README.md:40]: backtics seemed correct for these (and probably the items above as well) as these are code not text.

File: examples/lighting-app/bouffalolab/bl602/src/AppTask.cpp [hunk 2]
```diff
@@ -138,7 +138,9 @@ void AppTask::AppTaskMain(void * pvParameter)
     log_info("App Task entered\r\n");
 
     sWiFiNetworkCommissioningInstance.Init();
-    chip::Server::GetInstance().Init();
+    static chip::CommonCaseDeviceServerInitParams initParams;
+    (void) initParams.InitializeStaticResourcesBeforeServerInit();
```
[the remark this answers is not in the document]
Comment (jczhang777) [reply examples/lighting-app/bouffalolab/bl602/src/AppTask.cpp]: Thanks for pointing it out. I changed it.

File: examples/lighting-app/bouffalolab/bl602/README.md (hunk 1 above)
Comment (jczhang777) [reply examples/lighting-app/bouffalolab/bl602/README.md]: Thanks for pointing it out. I changed it

**Comment by jczhang777** [comment]
Thanks for pointing it out. Should I close this pr, and make a new pr to fix it.

**Comment by jczhang777** [comment]
> Please update summary with full information:
> 
>     * delete boilerplate (`What is being fixed? Examples:` seems not relevant as a text)
> 
>     * `Fix pbuf issue` - what issue?
> 
>     * `Update README` with what? for what purpose?
> 
> 
> Also add "how was this tested" section.

Thanks for pointing it out. Should I close this pr, and make a new pr to fix it.

**Comment by jczhang777** [comment]
> Please update summary with full information:
> 
>     * delete boilerplate (`What is being fixed? Examples:` seems not relevant as a text)
> 
>     * `Fix pbuf issue` - what issue?
> 
>     * `Update README` with what? for what purpose?
> 
> 
> Also add "how was this tested" section.

Thanks for pointing it out. Should I close this pr, and make a new pr to fix it?

**Comment by jczhang777** [comment]
> Please update summary with full information:
> 
>     * delete boilerplate (`What is being fixed? Examples:` seems not relevant as a text)
> 
>     * `Fix pbuf issue` - what issue?
> 
>     * `Update README` with what? for what purpose?
> 
> 
> Also add "how was this tested" section.

Thanks for point it out. I fixed it.Is there any inappropriate place that I need to modify

**Comment by andy31415** [comment]
@jczhang777 the summary now says that pbuf size was updated to match matter. Where is that update? I wanted to make sure that if 2 sizes need to be in sync, there is a comment explaining this.

is this the later SDK repo update or something else?

File: examples/lighting-app/bouffalolab/bl602/src/AppTask.cpp [hunk 3]
```diff
@@ -138,7 +138,9 @@ void AppTask::AppTaskMain(void * pvParameter)
     log_info("App Task entered\r\n");
 
     sWiFiNetworkCommissioningInstance.Init();
-    chip::Server::GetInstance().Init();
+    chip::CommonCaseDeviceServerInitParams initParams;
+    initParams.InitializeStaticResourcesBeforeServerInit();
+    chip::Server::GetInstance().Init(initParams);
```
Comment (andy31415) [inline examples/lighting-app/bouffalolab/bl602/src/AppTask.cpp:143]: Does Init return CHIP_ERROR? Could you handle it (at least log on error)?

**Review by andy31415 — APPROVED** [review]

**Comment by andy31415** [comment]
> @jczhang777 the summary now says that pbuf size was updated to match matter. Where is that update? I wanted to make sure that if 2 sizes need to be in sync, there is a comment explaining this.
> 
> is this the later SDK repo update or something else?

Guessing this is https://github.com/bouffalolab/bl_iot_sdk_matter/commit/3f19841e1048a9db6aeea0acab4b973767483028

I think summary should have stated that: this includes a SDK update for bouffalolab, which changes pbuf size from 96 back to 128. This fixes ....

File: examples/lighting-app/bouffalolab/bl602/src/AppTask.cpp (hunk 3 above)
Comment (jczhang777) [reply examples/lighting-app/bouffalolab/bl602/src/AppTask.cpp]: > @jczhang777 the summary now says that pbuf size was updated to match matter. Where is that update? I wanted to make sure that if 2 sizes need to be in sync, there is a comment explaining this.
> 
> is this the later SDK repo update or something else?

When lwIP PBUf alloc RAW, the length of PBUF_LINK_ENCAPSULATION_HLEN must be reserved 128 bytes that is sufficient.
I'm sorry for the trouble you by not describing clearly.

Comment (jczhang777) [reply examples/lighting-app/bouffalolab/bl602/src/AppTask.cpp]: I have solved the problem as you said. Thank you very much for your careful review. I'll pay attention to that next time.

**Pull request closed without merging**